allennlp/data/dataset_readers/dataset_reader.py

```diff
@@ -84,10 +85,8 @@ def cache_data(self, cache_directory: str) -> None:
         is responsible for calling this method and ensuring that unique parameters correspond to
         unique cache directories.  If you don't use our commands, that is your responsibility.
         """
-        if not cache_directory.endswith('/'):
-            cache_directory += '/'
-        self._cache_directory = cache_directory
-        os.makedirs(self._cache_directory, exist_ok=True)
+        self._cache_directory = pathlib.Path(cache_directory)
+        self._cache_directory.mkdir(exist_ok=True)
```

Comment:
This is not recursive, so it breaks in the normal use case.  You have to use `os.makedirs()` instead.  I _thought_ that `os.makedirs` would require the trailing slash, which is why I didn't want to remove it, but it appears to be working with `pathlib.Path(cache_directory)`.  I just pushed a change here; if it passes I'll merge this.

Reply:
oops, it looks like `Path.mkdir` takes a `parents=True` parameter that makes it recursive.